File: backend/seed.py
from datetime import date, timedelta

from app.database import Base, SessionLocal, engine
from app.models import (
    Ambassador,
    Benefit,
    Client,
    ContractTemplate,
    Cycle,
    AnamnesisTemplate,
    PrescriptionTemplate,
    Product,
    Procedure,
    Setting,
    User,
)
from app.services.program import build_coupon, build_link
from app.utils.migrations import ensure_sqlite_columns
from app.utils.security import get_password_hash


Base.metadata.create_all(bind=engine)
ensure_sqlite_columns()

benefits = [
    (3, "Limpeza de pele"),
    (5, "Peeling"),
    (7, "Microagulhamento"),
    (8, "Skinbooster"),
    (10, "Botox 1 região"),
    (12, "Lipo enzimática papada ou abdômen"),
    (14, "Botox 2 regiões"),
    (16, "Preenchimento labial ou preenchimento de bigode chinês"),
    (18, "Bioestimulador de colágeno, preenchimento ou combo botox + skinbooster"),
    (20, "Harmonização facial"),
]

procedures = [
    ("Limpeza de pele", "Facial", 180),
    ("Peeling químico", "Facial", 320),
    ("Microagulhamento", "Facial", 550),
    ("Skinbooster", "Injetáveis", 900),
    ("Botox", "Injetáveis", 750),
    ("Preenchimento labial", "Injetáveis", 1200),
]

settings = {
    "clinic_name": "Clínica TF",
    "clinic_cnpj": "00.000.000/0001-00",
    "default_credit_after_20": "50",
    "program_rules": "Cada indicação validada vale 1 ponto. Benefícios são preservados ao fim do ciclo.",
    "professionals": "Dra. Fernanda; Dra. Marina",
}


def upsert_setting(db, key, value):
    row = db.query(Setting).filter(Setting.key == key).first()
    if not row:
        row = Setting(key=key)
        db.add(row)
    row.value = value


def main():
    db = SessionLocal()
    try:
        if not db.query(User).filter(User.username == "admin").first():
            db.add(User(username="admin", hashed_password=get_password_hash("admin123")))

        for key, value in settings.items():
            upsert_setting(db, key, value)

        for points, name in benefits:
            if not db.query(Benefit).filter(Benefit.required_points == points, Benefit.name == name).first():
                db.add(Benefit(required_points=points, name=name, description=name, type="procedimento"))

        for name, category, price in procedures:
            if not db.query(Procedure).filter(Procedure.name == name).first():
                db.add(Procedure(name=name, category=category, description=f"Procedimento de {category.lower()}", default_price=price))

        if not db.query(Cycle).filter(Cycle.status == "ativo").first():
            today = date.today()
            db.add(Cycle(name="Ciclo Atual", start_date=today, end_date=today + timedelta(days=90), status="ativo", notes="Ciclo inicial de demonstração"))

        demo_client = db.query(Client).filter(Client.email == "ana@example.com").first()
        if not demo_client:
            demo_client = Client(full_name="Ana Ribeiro", phone="11999990000", email="ana@example.com", cpf="12345678900", origin="Cliente antiga", notes="Cliente demo")
            db.add(demo_client)
            db.flush()

        if not db.query(Ambassador).filter(Ambassador.client_id == demo_client.id).first():
            coupon = build_coupon(db, demo_client.full_name)
            db.add(Ambassador(client_id=demo_client.id, public_name="Ana R.", coupon=coupon, exclusive_link=build_link(coupon), status="ativa", level="comum"))

        if not db.query(ContractTemplate).first():
            db.add(ContractTemplate(
                name="Contrato padrão de procedimento",
                category="Procedimentos estéticos",
                content=(
                    "CONTRATO DE PRESTAÇÃO DE SERVIÇOS\n\n"
                    "Cliente: {{nome_cliente}}\nCPF: {{cpf_cliente}}\nTelefone: {{telefone_cliente}}\n"
                    "Procedimento: {{procedimento}}\nValor: R$ {{valor}}\nData: {{data}}\nProfissional: {{profissional}}\n\n"
                    "{{nome_clinica}} - CNPJ {{cnpj_clinica}}"
                ),
            ))

        if not db.query(Product).first():
            db.add_all([
                Product(name="Protetor solar FPS 50", category="Dermocosmético", sku="PROT-FPS50", price=89.90, cost=45, stock_quantity=12, min_stock=3),
                Product(name="Sérum hidratante", category="Dermocosmético", sku="SER-HID", price=129.90, cost=62, stock_quantity=8, min_stock=2),
            ])

        if not db.query(AnamnesisTemplate).first():
            db.add(AnamnesisTemplate(
                name="Anamnese estética padrão",
                category="Estética",
                content="Queixa principal:\nHistórico de alergias:\nMedicamentos em uso:\nProcedimentos anteriores:\nContraindicações:",
            ))

        if not db.query(PrescriptionTemplate).first():
            db.add(PrescriptionTemplate(
                name="Orientações pós-procedimento",
                content="Cliente: {{nome_cliente}}\nOrientações:\n- Evitar exposição solar.\n- Usar protetor solar.\n- Retornar em caso de reação inesperada.",
            ))

        db.commit()
        print("Seed concluído. Admin padrão: admin / admin123")
    finally:
        db.close()


if __name__ == "__main__":
    main()

File: clinicatf-next/prisma/seed.js
const { PrismaClient } = require('@prisma/client');
const bcrypt = require('bcryptjs');

const prisma = new PrismaClient();

async function main() {
  // Criar Usuário Admin
  const adminExists = await prisma.user.findUnique({
    where: { username: 'admin' },
  });

  if (!adminExists) {
    const hashedPassword = await bcrypt.hash('admin123', 10);
    await prisma.user.create({
      data: {
        username: 'admin',
        hashed_password: hashedPassword,
      },
    });
    console.log('Usuário admin criado (admin / admin123)');
  }

  // Configurações
  const settings = {
    clinic_name: 'Clínica TF',
    clinic_cnpj: '00.000.000/0001-00',
    default_credit_after_20: '50',
    program_rules: 'Cada indicação validada vale 1 ponto. Benefícios são preservados ao fim do ciclo.',
    professionals: 'Dra. Fernanda; Dra. Marina',
  };

  for (const [key, value] of Object.entries(settings)) {
    await prisma.setting.upsert({
      where: { key },
      update: { value },
      create: { key, value },
    });
  }
  console.log('Configurações criadas');

  // Benefícios
  const benefits = [
    { points: 3, name: 'Limpeza de pele' },
    { points: 5, name: 'Peeling' },
    { points: 7, name: 'Microagulhamento' },
    { points: 8, name: 'Skinbooster' },
    { points: 10, name: 'Botox 1 região' },
    { points: 12, name: 'Lipo enzimática papada ou abdômen' },
    { points: 14, name: 'Botox 2 regiões' },
    { points: 16, name: 'Preenchimento labial ou preenchimento de bigode chinês' },
    { points: 18, name: 'Bioestimulador de colágeno, preenchimento ou combo botox + skinbooster' },
    { points: 20, name: 'Harmonização facial' },
  ];

  for (const b of benefits) {
    const exists = await prisma.benefit.findFirst({
      where: { required_points: b.points, name: b.name },
    });
    if (!exists) {
      await prisma.benefit.create({
        data: {
          required_points: b.points,
          name: b.name,
          description: b.name,
          type: 'procedimento',
        },
      });
    }
  }
  console.log('Benefícios criados');

  // Procedimentos
  const procedures = [
    { name: 'Limpeza de pele', category: 'Facial', price: 180 },
    { name: 'Peeling químico', category: 'Facial', price: 320 },
    { name: 'Microagulhamento', category: 'Facial', price: 550 },
    { name: 'Skinbooster', category: 'Injetáveis', price: 900 },
    { name: 'Botox', category: 'Injetáveis', price: 750 },
    { name: 'Preenchimento labial', category: 'Injetáveis', price: 1200 },
  ];

  for (const p of procedures) {
    const exists = await prisma.procedure.findUnique({
      where: { name: p.name },
    });
    if (!exists) {
      await prisma.procedure.create({
        data: {
          name: p.name,
          category: p.category,
          description: `Procedimento de ${p.category.toLowerCase()}`,
          default_price: p.price,
        },
      });
    }
  }
  console.log('Procedimentos criados');

  // Ciclo Atual
  const activeCycle = await prisma.cycle.findFirst({
    where: { status: 'ativo' },
  });

  if (!activeCycle) {
    const today = new Date();
    const endDate = new Date(today);
    endDate.setDate(endDate.getDate() + 90);

    await prisma.cycle.create({
      data: {
        name: 'Ciclo Atual',
        start_date: today,
        end_date: endDate,
        status: 'ativo',
        notes: 'Ciclo inicial de demonstração',
      },
    });
    console.log('Ciclo criado');
  }

  // Cliente Demo
  let demoClient = await prisma.client.findFirst({
    where: { email: 'ana@example.com' },
  });

  if (!demoClient) {
    demoClient = await prisma.client.create({
      data: {
        full_name: 'Ana Ribeiro',
        phone: '11999990000',
        email: 'ana@example.com',
        cpf: '12345678900',
        origin: 'Cliente antiga',
        notes: 'Cliente demo',
      },
    });
    console.log('Cliente demo criado');
  }

  // Embaixadora Demo
  const ambassadorExists = await prisma.ambassador.findUnique({
    where: { client_id: demoClient.id },
  });

  if (!ambassadorExists) {
    const coupon = 'ANARIBEIRO';
    await prisma.ambassador.create({
      data: {
        client_id: demoClient.id,
        public_name: 'Ana R.',
        coupon: coupon,
        exclusive_link: `https://wa.me/5511900000000?text=Olá,%20tenho%20o%20cupom%20${coupon}`,
        status: 'ativa',
        level: 'comum',
      },
    });
    console.log('Embaixadora demo criada');
  }

  console.log('Seed completo!');
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
